packages/block-library/src/pattern/edit.js

```diff
@@ -18,9 +19,23 @@ const PatternEdit = ( { attributes, clientId } ) => {
 		[ attributes.slug ]
 	);
 
+	const currentThemeStylesheet = useSelect(
+		( select ) => select( coreStore ).getCurrentTheme().stylesheet
+	);
+
 	const { replaceBlocks, __unstableMarkNextChangeAsNotPersistent } =
 		useDispatch( blockEditorStore );
 
+	function injectThemeAttributeInBlockTemplateContent( block ) {
+		if (
+			block.name === 'core/template-part' &&
+			block.attributes.theme === undefined
+		) {
+			block.attributes.theme = currentThemeStylesheet;
+		}
+		return block;
+	}
```

Comment:
We should also search through the `innerBlocks` to find template parts that may also need the theme attribute. Something like this:

```suggestion
	function injectThemeAttributeInBlockTemplateContent( block ) {
		if (
			block.innerBlocks.find(
				( block ) => block.name === 'core/template-part'
			)
		) {
			block.innerBlocks = block.innerBlocks.map( ( block ) => {
				if (
					block.name === 'core/template-part' &&
					block.attributes.theme === undefined
				) {
					block.attributes.theme = currentThemeStylesheet;
				}
				return block;
			} );
		}

		if (
			block.name === 'core/template-part' &&
			block.attributes.theme === undefined
		) {
			block.attributes.theme = currentThemeStylesheet;
		}
		return block;
	}
```

Reply:
Thank you.  I have brought that change in with only minor adjustments and seems to work as intended.  Thank you for finding and fixing that.